app.js

```diff
@@ -0,0 +1,88 @@
+const path = require("path");
+const bodyParser = require("body-parser");
+const express = require("express");
+
+const app = express();
+const expressWs = require("express-ws")(app);
+
+const javaScriptUtils = require("./app/utils/javascript-utils/javascript-utils");
+const runtimeVariables = require("./configs/runtime-variables");
+const Settings = require("./app/settings/settings");
+
+const settings = new Settings();
+
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(bodyParser.json());
+
+app.use(express.static(path.join(__dirname, "resources")));
+app.use(express.static(path.join(__dirname, "scripts")));
+
+app.get("/settings/current", (req, res) => {
+  const currentSettings = settings.getCurrentSettings();
+
+  res.json({
+    currentSettings
+  });
+});
+
+app.post("/settings/onevent", (req, res) => {
+  settings.setOnEventSettings(req.body);
+
+  const currentSettings = settings.getCurrentSettings();
+
+  res.json({
+    success: true,
+    currentSettings
+  });
+});
+
+app.post("/settings/periodic", (req, res) => {
+  settings.setPeriodicSettings(req.body);
+
+  const currentSettings = settings.getCurrentSettings();
+
+  res.send({
+    success: true,
+    currentSettings
+  });
+});
+
+app.get("/disconnect", (req, res) => {
+  expressWs.getWss().clients.forEach(client => {
+    client.close();
+  });
+
+  res.json({
+    success: true
+  });
+});
+
+app.ws("/", ws => {
+  ws.on("message", () => {
+    if (
+      javaScriptUtils.objectIsNotEmpty(settings.perRequest.dummyResponseMessage)
+    ) {
+      ws.send(JSON.stringify(settings.perRequest.dummyResponseMessage));
+    }
+  });
+
+  if (
+    javaScriptUtils.objectIsNotEmpty(settings.periodic.dummyResponseMessage)
+  ) {
+    setInterval(() => {
+      // https://github.com/websockets/ws/issues/793
+      const isConnectionOpen = ws.readyState === ws.OPEN;
+
+      if (isConnectionOpen) {
+        ws.send(JSON.stringify(settings.periodic.dummyResponseMessage));
+      } else {
+        // It might not be necessary, but good to close the connection safely.
+        ws.close();
```

Comment:
I think this is redundant. Please test whether the server errors if you brute-force disconnect the client. If not, this can be removed.

Reply:
Yes, that's absolutely right, verified locally.